diff --git recipes/linux/linux-omap3_2.6.32.bb recipes/linux/linux-omap3_2.6.32.bb
index e6f4268..88f01ad 100644
--- recipes/linux/linux-omap3_2.6.32.bb
+++ recipes/linux/linux-omap3_2.6.32.bb
@@ -19,6 +19,12 @@ SRC_URI = "git://www.sakoman.com/git/linux-omap-2.6.git;branch=omap3-2.6.32;prot
 
 SRC_URI_append = " \
            file://${BOOT_SPLASH} \
+           file://1-omap2-mcspi-cleanup-null-ref.patch;patch=1 \
+           file://2-mcspi-plat-add-max-clk-div.patch;patch=1 \
+           file://3-omap2-mcspi-max-clk-div.patch;patch=1 \
+           file://4-omap2-mcspi-use-transfer-speed-override.patch;patch=1 \
+           file://5-omap2-mcspi-use-platdata-for-num-cs.patch;patch=1 \
+           file://6-omap2-mcspi-setup-transfer-reduce-logging.patch;patch=1 \
            "
 
 do_configure_prepend() {
